drivers/adc/adc_shell.c

```diff
@@ -32,17 +32,41 @@
 
 #define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
 #include <logging/log.h>
+
+#include <devicetree.h>
+
 LOG_MODULE_REGISTER(adc_shell);
 
-struct adc_hdl {
-	char *device_name;
-	struct adc_channel_cfg channel_config;
-	uint8_t resolution;
-};
+#define CMD_ACQ_TIME_HELP 			\
+	"Configure acquisition time."		\
+	"\nUsage: acq_time <time> <unit>"	\
+	"\nunits: us, ns, ticks\n"
+
+#define CMD_CH_ID_HELP				\
+	"Configure channel id\n"		\
+	"Usage: channel <channel_id>\n"
+
+#define CMD_PRINT_HELP				\
+	"Print current configuration"
+
+#define CMD_READ_HELP				\
+	"Read adc value\n"			\
+	"Usage: read <channel>\n"
 
+#define CMD_RES_HELP				\
+	"Configure resolution\n"		\
+	"Usage: resolution <resolution>\n"
+
+#define CMD_REF_HELP				\
+	"Configure resolution\n"
+
+#define CMD_HELP_GAIN				\
+	"Configure gain."
+
+#define NODE_LABELS(n) DT_INST_LABEL(n),
 #define ADC_HDL_LIST_ENTRY(inst)					\
 	{								\
-		.device_name = DT_INST_LABEL(inst),			\
+		.device_name = inst,					\
```

Comment:
If I reading it right (otherwise, please discard this comment):
While this is working and doing the expected thing, the use of `inst` is misleading here.
In DT macros, we're usually calling INST the ordinal number of the node (2nd, 3rd, ..).
In `ADC_HDL_LIST_ENTRY`, the argument is actually the label property value ("ADC_X" string).

So, to clarify this piece of code, I'd witch argument to something less misleading: `node_label` or similar.

Reply:
Sure, I will update the naming. It is a property label like `ADC_0` or `ADC_1`.